ui/components/toolbar.go

```diff
@@ -13,14 +14,31 @@ func NewToolBar(
 	pullRequestMenuItemClickedEventHandler critic.PullRequestMenuItemClickedEventHandler,
 	analyzeButtonHandler critic.AnalyzeButtonClickedEventHandler,
 ) fyne.CanvasObject {
+	selectedGPTModel := ""
+	gptModelSelect := widget.NewSelect(critic.SupportedGPTModels, func(selected string) {
+		selectedGPTModel = selected
+	})
+
+	// Wrapper function to check GPT model selection before executing the handler
+	executeWithModelCheck := func(handler func(string)) func() {
+		return func() {
+			if selectedGPTModel == "" {
+				dialog.ShowInformation("Info", "Please select a GPT model first.", fyne.CurrentApp().Driver().AllWindows()[0])
```

Comment:
Would this pattern for fetching the window work here?

https://github.com/kmesiab/ai-code-critic/blob/b04ac5358e914dc70ab3eaa93ce6604b427cbfda/main.go#L32

Reply:
<img width="791" alt="image" src="https://github.com/kmesiab/ai-code-critic/assets/34355192/84814e0b-71c0-472a-a577-c5f2fb8baff1">

yep